src/http/HttpResponse.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as types from '@azure/functions';
+import { HttpResponseInit } from '@azure/functions';
+import { Blob } from 'buffer';
+import { ReadableStream } from 'stream/web';
+import { FormData, Headers, Response as uResponse, ResponseInit as uResponseInit } from 'undici';
+import { isDefined } from '../utils/nonNull';
+
+export class HttpResponse implements types.HttpResponse {
+    readonly cookies: types.Cookie[];
+    readonly enableContentNegotiation: boolean;
+
+    #uRes: uResponse;
+
+    constructor(resInit?: HttpResponseInit) {
+        const uResInit: uResponseInit = { status: resInit?.status, headers: resInit?.headers };
+        if (isDefined(resInit?.jsonBody)) {
+            this.#uRes = uResponse.json(resInit?.jsonBody, uResInit);
+        } else {
+            this.#uRes = new uResponse(resInit?.body, uResInit);
+        }
+
+        this.cookies = resInit?.cookies || [];
+        this.enableContentNegotiation = !!resInit?.enableContentNegotiation;
+    }
+
+    get status(): number {
+        return this.#uRes.status;
+    }
+
+    get headers(): Headers {
+        return this.#uRes.headers;
+    }
+
+    get body(): ReadableStream<any> | null {
+        return this.#uRes.body;
+    }
+
+    get bodyUsed(): boolean {
+        return this.#uRes.bodyUsed;
+    }
+
+    async arrayBuffer(): Promise<ArrayBuffer> {
+        return await this.#uRes.arrayBuffer();
+    }
+
+    async blob(): Promise<Blob> {
+        return await this.#uRes.blob();
+    }
+
+    async formData(): Promise<FormData> {
+        return await this.#uRes.formData();
+    }
+
+    async json(): Promise<unknown> {
+        return await this.#uRes.json();
+    }
+
+    async text(): Promise<string> {
+        return await this.#uRes.text();
+    }

```

Comment:
Do these need to do the async/await themselves, or could you just return the Promise, like so:

```suggestion
    blob(): Promise<Blob> {
        return this.#uRes.blob();
    }

    formData(): Promise<FormData> {
        return this.#uRes.formData();
    }

    json(): Promise<unknown> {
        return this.#uRes.json();
    }

    text(): Promise<string> {
        return this.#uRes.text();
    }

```

Reply:
My concern was forgetting to await a return value in a `try/catch` which is why I made it a habit to add `await` everywhere. But I found a [lint rule](https://typescript-eslint.io/rules/return-await/) that can handle that situation. I'll turn on the lint rule in a separate PR as it might affect a lot of files